helpers/internal/waitForElem.js

```diff
@@ -0,0 +1,47 @@
+import { getComposedChildren } from '../dom.js';
+
+export async function waitForElem(elem, predicate = () => true) {
```

Comment:
Worth noting that the predicate isn't currently invoked for the first/root `elem` passed in. If the consumer were generically iterating over an array of elements, we'd probably want to check.

If we _do_ check the predicate for the element passed, the consumer needs to be careful which element they pass and what their predicate is.